go/api/config.go

```diff
@@ -276,8 +276,15 @@ func (config *GlideClientConfiguration) WithReadFrom(readFrom ReadFrom) *GlideCl
 // encompasses sending the request, awaiting for a response from the server, and any required reconnections or retries. If the
 // specified timeout is exceeded for a pending request, it will result in a timeout error. If not set, a default value will be
 // used.
+//
+// WARNING: If requestTimeout exceeds the max duration of 2^32 - 1, a default value will be used.
 func (config *GlideClientConfiguration) WithRequestTimeout(requestTimeout time.Duration) *GlideClientConfiguration {
-	config.requestTimeout = requestTimeout
+	// Convert back to milliseconds and check if requestTimeout exceeds bounds of a uint32
+	milliseconds := requestTimeout.Milliseconds()
+	if milliseconds < 0 || milliseconds > (1<<32-1) {
+		milliseconds = 5000
```

Comment:
Why not allow the rust client-core implementation to use whatever timeout default it would use if `WithRequestTimeout` had never been called.